src/main/java/catan/services/util/play/MainStageUtil.java

```diff
@@ -30,6 +36,45 @@ public void updateNextMove(GameBean game) {
         game.setCurrentMove(nextMoveNumber);
     }
 
+    public void distributeResources(GameBean game) {
+        Integer diceValue = game.getDiceFirstValue() != null ? game.getDiceFirstValue() + game.getDiceSecondValue() : 0;
+        for (HexBean hex : game.getHexes()) {
+            if (diceValue.equals(hex.getDice())) {
+                for (NodeBean node : hex.getNodes().listAllNotNullItems()) {
+                    if (node.getBuilding() != null) {
+                        giveResourceToPlayer(node.getBuilding().getBuildingOwner(), node.getBuilding().getBuilt(), hex.getResourceType());
+                    }
```

Comment:
there can be even one cycle-condition, but 6 tabs is not a good code. If you don't have any ideas, leave it as is